From ef551a7d3abfedb5021db92c2eb58b4aade81214 Mon Sep 17 00:00:00 2001
From: Robert Crossfield <robcrossfield@gmail.com>
Date: Sat, 16 Mar 2024 15:13:39 +1100
Subject: [PATCH 35/50] Fix for when music pointer not set

---
 Source/Amiga/Sound_Amiga.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/Source/Amiga/Sound_Amiga.cpp b/Source/Amiga/Sound_Amiga.cpp
index 4d3d936..c87c189 100644
--- a/Source/Amiga/Sound_Amiga.cpp
+++ b/Source/Amiga/Sound_Amiga.cpp
@@ -246,7 +246,7 @@ void cSound_Amiga::Music_Stop() {
 }
 
 void cSound_Amiga::Music_SetVolume(int16 pChannel, int16 pVolume) {
-	if (mCurrentMusic == 0)
+	if (!mCurrentMusic)
 		return;
 
 	mCurrentMusic->volumeMax = pVolume;
@@ -255,6 +255,9 @@ void cSound_Amiga::Music_SetVolume(int16 pChannel, int16 pVolume) {
 }
 
 int16 cSound_Amiga::Music_GetVolume(int16 pChannel) {
+	if (!mCurrentMusic)
+		return 0;
+
 	return mCurrentMusic->volumeMax;
 	//auto channel = mCurrentMusic->getChannel(pChannel);
 	//return channel->volume;
-- 
2.25.1

